### DISCUSSION TITLE
How to prevent the meta json file from each page to contain the locale messages?

### DISCUSSION THREAD

**Discussion by iKonrad** [vercel/next.js#25153] [Help]
So I'm working on getting the integration with react-intl.

Currently I've made a custom `App` with `getInitialProps` that loads the locales dynamically and passes it as props.

The problem is that the locale files are quite big, and `getInitialProps` passes it to every page as props, which increases the size of the page meta json files unnecessarily (even if the language of the next page is the same and all the locales are loaded).

Is there a way to mitigate it and don't include the locales in json meta?

**Comment by joggienl** [comment]
I would like to bump this question.

Has anyone came up with some solution to pass _all_ translations for every page every time?

With all translations I mean 1 language.

For instance, a homepage of an e-commerce site with a members area will load all translations for the members only pages also on the homepage.
